features/admin.userstores.v1/api/use-get-user-store-meta-data-type.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import useRequest, {
+    RequestConfigInterface,
+    RequestErrorInterface,
+    RequestResultInterface
+} from "@wso2is/admin.core.v1/hooks/use-request";
+import { store } from "@wso2is/admin.core.v1/store";
+import { HttpMethods } from "@wso2is/core/models";
+import { QueryParams, UserstoreType } from "../models/user-stores";
+
+/**
+ * Retrieves meta data of a single userstore type.
+ *
+ * @param id - The ID of the userstore type.
+ * @param params - Optional query params such as limit, offset, filter, sort, attributes, etc.
+ * @param shouldFetch - Whether the request should be made.
+ *
+ * @returns A RequestResultInterface containing data, error, isLoading, etc.
+ */
+export const useGetUserStoreMetaDataType = <
+    Data = UserstoreType,
+    Error = RequestErrorInterface
+    >(
+        id: string,
+        params?: QueryParams,
+        shouldFetch: boolean = true
+    ): RequestResultInterface<Data, Error> => {
+
+    // Build the request configuration.
+    const requestConfig: RequestConfigInterface = {
+        headers: {
+            Accept: "application/json",
+            "Access-Control-Allow-Origin": store.getState().config.deployment.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        params,
+        url: `${store.getState().config.endpoints.userStores}/meta/types/${ id }`
+    };
+
+    // Pass the config to `useRequest`, or null if we don't want to fetch yet.
+    const {
+        data,
+        error,
+        isValidating,
+        mutate,
+        response
+    } = useRequest<Data, Error>(shouldFetch ? requestConfig : null);
+
+    return {
+        data,
+        error,
+        isLoading: !error && !data,
```

Comment:
SWR returns `isLoading` property

```suggestion
        isLoading,
```

Reply:
Fixed with f21ea57e816da2c17eabe26fd5a9cc88adec10e4